src/tests/context.test.js

```diff
@@ -36,6 +36,135 @@ describe('context', () => {
     });
   });
 
+  describe('persistence', () => {
+    let context;
+    beforeEach(() => {
+      context = new Context();
+      context.initialize('test-persistence', {}, {});
+      // Clear any existing stored data
+      if (typeof localStorage !== 'undefined') {
+        localStorage.clear();
+      }
+      if (typeof sessionStorage !== 'undefined') {
+        sessionStorage.clear();
+      }
+    });
+
+    it('should configure persistence for a key', () => {
+      expect(() => context.setPersistence('user.preference', STORAGE_TYPE_USER)).to.not.throw();
```

Comment:
I can't remember if keys are treated as flattened paths or are hierarchical. Nevertheless, the dot-delimited key name might lead to confusion. If I were, say, to set persistence on `user`, should that also cover `user.preference`?(Under the current implementation it wouldn't.)

Reply:
The current current implementation supports dot-delimited. This commit is just following the current pattern.